src/components/modules/settings/Settings.tsx

```diff
@@ -0,0 +1,40 @@
+import { Dispatch, FC, SetStateAction } from 'react'
+import { DifficultySelector } from '../difficulty/DifficultySelector'
+import styles from './Settings.module.scss'
+import { Difficulty } from '../../../constants'
+import { LayerSelector } from '../layer/LayerSelector'
+
+interface SettingsProps {
+  activeDifficulty: Difficulty
+  setActiveDifficulty: Dispatch<SetStateAction<Difficulty>>
+  handleResetClick: () => void
+  filteredLayer?: number
+  totalLayers?: number
+  setFilteredLayer?: (layer: number) => void
+}
+export const Settings: FC<SettingsProps> = ({
```

Comment:
It seems `React.FC` used to be the recommended way to add typing, but with React 18 and TS 5, the inferred way is more commonly used:
```tsx
export const Settings = ({ ... }: SettingsProps) => { ... };
```

Source: https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components/

If it's not too much trouble, I'd use this style instead.

Reply:
Oh interesting, thanks for the link!